middle_end/closure/closure.ml

```diff
@@ -1058,22 +1058,29 @@ let rec close ({ backend; fenv; cenv ; mutable_vars } as env) lam =
               None ubody),
        approx)
   (* Compile-time constants *)
-  | Lprim(Pctconst c, [arg], _loc) ->
-      let cst, approx =
-        match c with
-        | Big_endian -> make_const_bool B.big_endian
-        | Word_size -> make_const_int (8*B.size_int)
-        | Int_size -> make_const_int (8*B.size_int - 1)
-        | Max_wosize -> make_const_int ((1 lsl ((8*B.size_int) - 10)) - 1 )
-        | Ostype_unix -> make_const_bool (Config.target_os_type = "Unix")
-        | Ostype_win32 -> make_const_bool (Config.target_os_type = "Win32")
-        | Ostype_cygwin -> make_const_bool (Config.target_os_type = "Cygwin")
-        | Backend_type ->
-            make_const_int 0 (* tag 0 is the same as Native here *)
+  | Lprim(Pctconst c, [arg], loc) ->
+      let cst f v =
+        let cst, approx = f v in
+        let arg, _approx = close env arg in
+        let id = Ident.create_local "dummy" in
+        Ulet(Immutable, Pgenval, VP.create id, arg, cst), approx
       in
-      let arg, _approx = close env arg in
-      let id = Ident.create_local "dummy" in
-      Ulet(Immutable, Pgenval, VP.create id, arg, cst), approx
+      begin match c with
+      | Big_endian -> cst make_const_bool B.big_endian
+      | Word_size -> cst make_const_int (8*B.size_int)
+      | Int_size -> cst make_const_int (8*B.size_int - 1)
+      | Max_wosize -> cst make_const_int ((1 lsl ((8*B.size_int) - 10)) - 1)
+      | Ostype_unix -> cst make_const_bool (Config.target_os_type = "Unix")
```

Comment:
Sigh. TIL that the split between Windows and Unix are plumbed all the way down to closures. Hopefully this is just for some internal flambda optimization. (Unrelated to this PR, but just something I'll have to research later since it sounds like a barrier for Windows<->Unix cross-compilation.)